packages/simcore-sdk/src/simcore_sdk/node_ports_v2/serialization_v2.py

```diff
@@ -3,14 +3,21 @@
 from pprint import pformat
 from typing import Any, Dict, Set
 
+import pydantic
 from aiopg.sa.result import RowProxy
 from models_library.projects_nodes import NodeID
 from models_library.utils.nodes import compute_node_hash
+from packaging import version
 
 from ..node_ports.dbmanager import DBManager
 from ..node_ports.exceptions import InvalidProtocolError
 from .nodeports_v2 import Nodeports
 
+# NOTE: Keeps backwards compatibility with pydantic
+#   - from 1.8, it DOES NOT need __root__ specifiers when nested
+_PYDANTIC_NEEDS_ROOT_SPECIFIED = version.parse(pydantic.VERSION) < version.parse("1.8")
```

Comment:
Why youse this? When you install simcore_sdk, you will require pydantic's version inside this package and result in collisions.

Reply:
- pydantic > 1.8 has some breaking changes.
- simcore-sdk is a library that requires pydantic (i.e. we do not impose strong constraints)
- services installing ``simcore-sdk`` determine which version to use
- simcore-sdk keeps backwards compatibility by including two code paths depending on the version of pydantic
- all services are happy

I do not understand what you mean by "collisions" ...

Reply:
When you install `simcore-sdk` you require these [packages/simcore-sdk/requirements/_base.txt#L108](https://github.com/ITISFoundation/osparc-simcore/blob/33e68f7912976603343fdfed855928af1db84276/packages/simcore-sdk/requirements/_base.txt#L108) That's what I mean. You already have 1.8 installed. What am I missing here?

Reply:
@GitHK : following your comment.

 All ``*.txt``-like requirements folder in any project UNDER ``packages`` are effectively **just to test against "the latest versions of everything"** (more details in ``/home/crespo/devp/osparc-simcore/requirements/python-dependencies.md``) ... but the real package "requirements" are in ``_base.in``. 

When we include this package in a service, the contraints on the service are determined by ``_base.in`` ... therefore there is no-collision. Have a look at [web/server/requirements.in](https://github.com/ITISFoundation/osparc-simcore/blob/fdd28e2d7edd9ae65955dbc23538f81ce584ba25/services/web/server/requirements/_base.in#L5)

TIP: To understand better the latter, open ``packages/service-library/setup.py`` and see how ``install_requires`` attribute is filled.

Reply:
Perfect, I've missed that one. It is now clear to me.